pkg/operation/shoot/shoot.go

```diff
@@ -149,7 +149,11 @@ func (b *Builder) WithDefaultDomains(defaultDomains []*gardenerutils.Domain) *Bu
 
 // Build initializes a new Shoot object.
 func (b *Builder) Build(ctx context.Context, c client.Reader) (*Shoot, error) {
-	shoot := &Shoot{}
+	var (
+		shoot    = &Shoot{}
+		secret   *corev1.Secret
```

Comment:
if you move initialization of `shoot.Secret` inside the if as suggested by @shafeeqes  then you also do not need to declare a variable for secret. When calling `gardenerutils.ConstructExternalDomain` you can get the secret from shoot.Secret